pkg/controller/container-runtime-config/helpers.go

```diff
@@ -954,3 +955,92 @@ func generateSigstoreRegistriesdConfig(clusterScopePolicies map[string]signature
 	}
 	return data, nil
 }
+
+func addScopeMirrorsSigstoreRegistriesdConfig(registriesDockerConfig map[string]dockerConfig, scope string, icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy, idmsRules []*apicfgv1.ImageDigestMirrorSet, itmsRules []*apicfgv1.ImageTagMirrorSet, sigstoreAttachment dockerConfig) {
+	type sourceMirrors struct {
+		source  string
+		mirrors []string
+	}
+
+	processRulesElement := func(rule []sourceMirrors) bool {
+		scopeIsSource := false
+		for _, r := range rule {
+			if strings.HasPrefix(r.source, scope) {
+				for _, mirror := range r.mirrors {
+					registriesDockerConfig[mirror] = sigstoreAttachment
+				}
+				if r.source == scope {
+					scopeIsSource = true
+				}
+			}
+		}
+		return scopeIsSource
+	}
+
+	findLongestPrefixSource := func(rule []sourceMirrors) []string {
```

Comment:
(I’d prefer minimizing the amount of `registries.conf` semantics which is hard-coded in MCO.)

---

I think this, when called from the 3 separate ICSP/IDMS/ITMS data sources, might find 3 different “best matches”; real `registries.conf` processing would only find one, possibly with mirrors merged from the 3.